src/dataset-builder/ConceptSearch.ts

```diff
@@ -0,0 +1,150 @@
+import { ActionBar } from 'src/components/ActionBar';
+import { Link, spinnerOverlay } from 'src/components/common';
+import { icon } from 'src/components/icons';
+import { TextInput, withDebouncedChange } from 'src/components/input';
+import { SimpleTable } from 'src/components/table';
+import { tableHeaderStyle } from 'src/dataset-builder/ConceptSelector';
+import { BuilderPageHeader } from 'src/dataset-builder/DatasetBuilderHeader';
+import { GetConceptsResponse, HighlightConceptName } from 'src/dataset-builder/DatasetBuilderUtils';
+import { DataRepo, SnapshotBuilderConcept as Concept, SnapshotBuilderDomainOption } from 'src/libs/ajax/DataRepo';
+import { useLoadedData } from 'src/libs/ajax/loaded-data/useLoadedData';
+import colors from 'src/libs/colors';
+
+type ConceptSearchProps = {
+  readonly initialSearch: string;
+  readonly domainOption: SnapshotBuilderDomainOption;
+  readonly onCancel: () => void;
+  readonly onCommit: (selected: Concept[]) => void;
+  readonly onOpenHierarchy: (
+    domainOption: SnapshotBuilderDomainOption,
+    selected: Concept[],
+    searchText: string
+  ) => void;
+  readonly actionText: string;
+  readonly datasetId: string;
+  readonly initialCart: Concept[];
+};
+
+const DebouncedTextInput = withDebouncedChange(TextInput);
+export const ConceptSearch = (props: ConceptSearchProps) => {
+  const { initialSearch, domainOption, onCancel, onCommit, onOpenHierarchy, actionText, datasetId, initialCart } =
+    props;
+  const [search, setSearch] = useState<string>(initialSearch);
+  const [cart, setCart] = useState<Concept[]>(initialCart);
+  const [concepts, searchConcepts] = useLoadedData<GetConceptsResponse>();
+
+  useEffect(() => {
+    void searchConcepts(() => {
+      return DataRepo().dataset(datasetId).searchConcepts(domainOption.root, search);
+    });
+  }, [search, datasetId, domainOption.root, searchConcepts]);
+  const tableLeftPadding = { paddingLeft: '2rem' };
+
+  return h(Fragment, [
+    h(BuilderPageHeader, [
+      h2({ style: { display: 'flex', alignItems: 'center' } }, [
+        h(
+          Link,
+          {
+            onClick: onCancel,
+            'aria-label': 'cancel',
+          },
+          [icon('left-circle-filled', { size: 32 })]
+        ),
+        div({ style: { marginLeft: 15 } }, [domainOption.category]),
+      ]),
+      div({ style: { position: 'relative' } }, [
+        h(DebouncedTextInput, {
+          onChange: (value: string) => {
+            setSearch(value);
+          },
+          value: search,
+          placeholder: 'Search',
+          type: 'search',
+          style: {
+            borderRadius: 25,
+            borderColor: colors.dark(0.2),
+            width: '100%',
+            maxWidth: 575,
+            height: '3rem',
+            marginRight: 20,
+            paddingLeft: 40,
+          },
+        }),
+        icon('search', { size: 18, style: { position: 'absolute', left: 15, top: '50%', marginTop: -9 } }),
```

Comment:
Nit: How would you feel about binding the icon size to a variable/const, and then setting `marginTop: iconSize / -2`? I think that might be safer if we end up change the icon size, and explain the margin more clearly